server/services/geocodes.js

```diff
@@ -0,0 +1,46 @@
+const { dbClient, collections } = require('../db');
+
+const getPointsFromPostalCodes = async (postalCodes) => {
+  const queryPoint = async (postalCode) => {
+    const spacedCode = `${postalCode.substring(0, 3)} ${postalCode.substring(3)}`;
+    const request = await dbClient.db[collections.GEOCODES].findOne({
+      postal_code: spacedCode,
+    });
+
+    if (request) {
+      // Happy path, full match on postal code
+      return {
+        lat: request.latitude,
+        lng: request.longitude,
+        match: request.postal_code.replace(' ', ''),
+      };
+    }
+    const approxRequest = await dbClient.db[collections.GEOCODES].findOne({
+      'postal_code like': `${spacedCode.slice(0, -1)}%`,
+    });
```

Comment:
if the initial request returned null, slice off the last digit and see if a nearby postal code exists in the database.